string direction
string state
float32 duration